### PR TITLE
fix(ls): reintroduce #9113

### PR DESCRIPTION
**Pull request by ajbt200128**
This PR adds support for libev again. See #9113 for the original PR

We need to wait for https://github.com/Homebrew/homebrew-core/pull/153184 before merging this PR
## Test plan
```
brew edit semgrep
```
then paste the contents of the formula from [here](https://github.com/Homebrew/homebrew-core/pull/153184), change
`head "https://github.com/returntocorp/semgrep.git", branch: "develop"`
to
`head "https://github.com/returntocorp/semgrep.git", branch: "austin/lsp-fd-issue-pt2"`
then `brew install --HEAD --debug semgrep`

### CODE REVIEW

**Comment by nmote** [comment]
I'll take a look at this in a bit more detail later if nobody else gets to it. Overall the test plan looks good, but given the ongoing issues with the release could you hold off until that is sorted out in case we need to cut a new branch?

**Comment by ajbt200128** [comment]
Yea this can wait

File: Makefile [hunk 1]
```diff
@@ -320,7 +320,7 @@ install-deps-ALPINE-for-pysemgrep:
 # - pkg-config?
 # - coreutils?
 # - gettext?
-BREW_DEPS=pcre gmp pkg-config coreutils gettext
+BREW_DEPS=pcre gmp pkg-config coreutils gettext libev
```
Comment (aryx) [inline Makefile:346]: there is an UBUNTU_DEPS also now in this Makefile you could also update.

File: Makefile [hunk 2]
```diff
@@ -347,7 +347,7 @@ homebrew-setup:
 	# See details at https://github.com/Homebrew/homebrew-core/pull/82693.
 	# This workaround may no longer be necessary.
 	opam install -y --deps-only --no-depexts ./libs/ocaml-tree-sitter-core
-	opam install -y --deps-only --no-depexts ./
+	LIBRARY_PATH="/opt/homebrew/lib" opam install -y --deps-only --no-depexts ./
```
Comment (aryx) [inline Makefile:373]: here it is fine.

File: Makefile [hunk 3]
```diff
@@ -320,7 +320,7 @@ install-deps-ALPINE-for-pysemgrep:
 # - pkg-config?
 # - coreutils?
 # - gettext?
-BREW_DEPS=pcre gmp pkg-config coreutils gettext
```
Comment (aryx) [inline Makefile:346]: add a coupling: comment saying to also update setup-m1-buiilder.sh
(actually would be better if setup-m1-builder.sh would use make to install the packages.

File: scripts/setup-m1-builder.sh [hunk 4]
```diff
@@ -17,7 +17,7 @@ sudo chmod g+rwx /usr/local/lib/
 
 brew update # Needed to sidestep bintray brownout
 # Usually we install python 3.7 for building wheels, but 3.7 isn't supported on m1 macs
-brew install opam pkg-config coreutils python pcre
+brew install opam pkg-config coreutils python pcre libev
```
Comment (aryx) [inline scripts/setup-m1-builder.sh]: would be better to call make, to avoid defining at yet another place a list of package to install

**Review by aryx — APPROVED** [review]

**Comment by aryx** [comment]
Nice test plan!
Maybe we should add those instructions somewhere.

**Comment by ajbt200128** [comment]
@nmote good to go?

**Comment by nmote** [comment]
Yes, the release is done! I haven't reviewed this specifically, though.

File: Makefile (hunk 1 above)
Comment (ajbt200128) [reply Makefile:346]: will do

File: Makefile (hunk 3 above)
Comment (ajbt200128) [reply Makefile:346]: going top change it to that

File: scripts/setup-m1-builder.sh (hunk 4 above)
Comment (ajbt200128) [reply scripts/setup-m1-builder.sh]: will do

**Comment by ajbt200128** [comment]
closing because weird CI

**Comment by ajbt200128** [comment]
REPOPEN

**Pull request merged**